packages/web-client/app/models/animated-workflow.ts

```diff
@@ -0,0 +1,244 @@
+  @reads('model.completedDetail') declare completedDetail: string;
+  @reads('model.title') declare title: string;
+  @tracked isCompletionRevealed = false;
+  @reads('postableCollection.visiblePostables')
+  declare visiblePostables: WorkflowPostable[];
+  postableCollection: AnimatedPostableCollection;
+
+  constructor(model: Milestone) {
+    this.model = model;
+    this.postableCollection = new AnimatedPostableCollection(
+      model.postableCollection
+    );
+  }
+
+  revealNext(): RevealResult {
+    let result = this.postableCollection.revealNext();
+    if (!result.revealed && !this.isCompletionRevealed) {
+      this.isCompletionRevealed = true;
+      result.revealed = true;
+    }
+    return result;
+  }
+
+  get isComplete(): boolean {
+    return this.postableCollection.isComplete && this.isCompletionRevealed;
+  }
+}
+
+class AnimatedPostableCollection {
+  model: PostableCollection;
+  @tracked revealPointer: WorkflowPostable | undefined;
+
+  constructor(model: PostableCollection) {
+    this.model = model;
+  }
+
+  get visiblePostables(): WorkflowPostable[] {
+    let result = [] as WorkflowPostable[];
+    if (!this.revealPointer) return result;
+    for (const postable of this.model.visiblePostables) {
+      result.push(postable);
+      if (postable === this.revealPointer) {
+        break;
+      }
+    }
+    return result;
+  }
+
+  get allPostablesVisible() {
+    return (
+      this.model.allNecessaryPostablesVisible &&
+      this.model.peekAtVisiblePostables().length ===
+        this.visiblePostables.length
+    );
+  }
+
+  revealNext(): RevealResult {
+    let postables = this.model.visiblePostables;
+    let index = this.revealPointer
+      ? postables.indexOf(this.revealPointer!)
+      : -1;
+    if (index === postables.length - 1) {
+      return {
+        revealed: false,
+      };
+    }
+    this.revealPointer = postables[index + 1];
+    return {
+      revealed: true,
+      postable: this.revealPointer,
+    };
+  }
+
+  get isComplete() {
+    if (!this.model.isComplete) {
+      return false;
+    }
+    let postables = this.model.visiblePostables;
+    let index = this.revealPointer
+      ? postables.indexOf(this.revealPointer!)
+      : -1;
+    return index === postables.length - 1;
+  }
+}
+
+export default class AnimatedWorkflow {
+  model: Workflow;
+  epilogue: AnimatedPostableCollection;
+  cancelationMessages: AnimatedPostableCollection;
+  milestones: AnimatedMilestone[];
+
+  constructor(model: Workflow) {
+    this.model = model;
+    this.model.animatedWrapper = this;
+    this.epilogue = new AnimatedPostableCollection(model.epilogue);
+    this.cancelationMessages = new AnimatedPostableCollection(
+      model.cancelationMessages
+    );
+    this.milestones = model.milestones.map((m) => new AnimatedMilestone(m));
+    taskFor(this.tickerTask).perform();
+  }
+  @reads('model.name') declare name: string;
+
+  @task
+  *tickerTask() {
+    this.startTestWaiter();
```

Comment:
I don't understand why we are starting the waiter once, and stopping it many times.